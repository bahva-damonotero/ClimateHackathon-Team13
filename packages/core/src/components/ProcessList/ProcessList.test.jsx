import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { ProcessList, ProcessListItem, ProcessListItemHeader } from '.';

describe('ProcessList', () => {
  const jsx = (
    <ProcessList>
      <ProcessListItem>
        <ProcessListItemHeader>Start a process.</ProcessListItemHeader>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo,
        ipsum sed pharetra gravida, orci magna rhoncus neque.
        <ul>
          <li>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi
            commodo, ipsum sed pharetra gravida, orci magna rhoncus nequem id
            pulvinar odio lorem non turpis.
          </li>
          <li>
            Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat
            conndimentum.
          </li>
          <li>Aliquam erat volutpat. Sed quis velit.</li>
        </ul>
      </ProcessListItem>

      <ProcessListItem>
        <ProcessListItemHeader>
          Proceed to the second step.
        </ProcessListItemHeader>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo,
        ipsum sed pharetra gravida, orci magna rhoncus neque, id pulvinar odio
        lorem non turpis. Nullam sit amet enim. Suspendisse id velit vitae
        ligula volutpat condimentum. Aliquam erat volutpat. Sed quis velit.
        Nulla facilisi. Nulla libero. Vivamus pharetra posuere sapien.
      </ProcessListItem>

      <ProcessListItem>
        <ProcessListItemHeader>
          Complete the step-by-step process.
        </ProcessListItemHeader>
        Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat
        condimentum. Aliquam erat volutpat. Sed quis velit. Nulla facilisi.
        Nulla libero. Vivamus pharetra posuere sapien.
      </ProcessListItem>
    </ProcessList>
  );

  it('should render correctly', () => {
    const { container } = render(jsx);

    expect(container.querySelector('.usa-process-list')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(jsx).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
