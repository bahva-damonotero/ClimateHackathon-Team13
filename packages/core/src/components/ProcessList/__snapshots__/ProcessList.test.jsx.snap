// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessList should match snapshot 1`] = `
<ol
  className="usa-process-list"
>
  <li
    className="usa-process-list__item"
  >
    <h4
      className="usa-process-list__heading"
    >
      Start a process.
    </h4>
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo, ipsum sed pharetra gravida, orci magna rhoncus neque.
    <ul>
      <li>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo, ipsum sed pharetra gravida, orci magna rhoncus nequem id pulvinar odio lorem non turpis.
      </li>
      <li>
        Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat conndimentum.
      </li>
      <li>
        Aliquam erat volutpat. Sed quis velit.
      </li>
    </ul>
  </li>
  <li
    className="usa-process-list__item"
  >
    <h4
      className="usa-process-list__heading"
    >
      Proceed to the second step.
    </h4>
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo, ipsum sed pharetra gravida, orci magna rhoncus neque, id pulvinar odio lorem non turpis. Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat condimentum. Aliquam erat volutpat. Sed quis velit. Nulla facilisi. Nulla libero. Vivamus pharetra posuere sapien.
  </li>
  <li
    className="usa-process-list__item"
  >
    <h4
      className="usa-process-list__heading"
    >
      Complete the step-by-step process.
    </h4>
    Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat condimentum. Aliquam erat volutpat. Sed quis velit. Nulla facilisi. Nulla libero. Vivamus pharetra posuere sapien.
  </li>
</ol>
`;
